bin/kill/kill.c

```diff
@@ -72,12 +71,7 @@ main(int argc, char *argv[])
 		if (argc == 1) {
 			if (!isdigit(**argv))
 				usage();
-			numsig = strtol(*argv, &ep, 10);
-			if (!**argv || *ep)
-				errx(2, "illegal signal number: %s", *argv);
-			if (numsig >= 128)
-				numsig -= 128;
-			if (numsig <= 0 || numsig >= sys_nsig)
+			if (str2sig(*argv, &numsig) < 0)
 				nosig(*argv);
 			printf("%s\n", sys_signame[numsig]);
```

Comment:
This change seems incorrect. Things like `kill -l INT` aren't supposed to work, while it is supposed to be possible to do something like `kill -l 139` and see `SEGV` (this is so that the signal name can be obtained from a value from `$?`). This change is likely to break various tests from bin/sh including builtins/kill2.0.

A change that could be made instead would be to use `sig2str` instead of `sys_signame` access.

An issue in the existing code is that the `long` return value from `strtol` is assigned to `int numsig`.

Reply:
> An issue in the existing code is that the `long` return value from `strtol` is assigned to `int numsig`.

For this I need to use strtonum as intended before.

Reply:
> This change seems incorrect. Things like `kill -l INT` aren't supposed to work

Why not?  It works in Bash at least.

> while it is supposed to be possible to do something like `kill -l 139` and see `SEGV`

This is working.

> A change that could be made instead would be to use `sig2str` instead of `sys_signame` access.

Will look into this.

Reply:
Restored the functionality with numeric values above 128 being special in `kill`.